cli/coredump/coredump.go

```diff
@@ -0,0 +1,212 @@
+package coredump
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"embed"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/apex/log"
+)
+
+//go:embed scripts/*
+var corescripts embed.FS
+
+// readEmbedFile reads content of embed file.
+func readEmbedFile(fs embed.FS, path string) (string, error) {
+	content, err := fs.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return string(content), nil
+}
+
+// writeScript writes content of embed file into file.
+func writeScript(corescripts embed.FS, path string) error {
+	content, err := readEmbedFile(corescripts, path)
+	if err != nil {
+		return err
+	}
+	path, err = os.Getwd()
+	err = ioutil.WriteFile(path+"/script.sh", []byte(content), 0770)
+	return err
+}
+
+// ExtractTarGz extracts tar.gz archive
+func ExtractTarGz(tarName string) error {
+	path, err := filepath.Abs(tarName)
+	if err != nil {
+		return err
+	}
+	archive, err := os.Open(path)
```

Comment:
You do the same loop with preparation twice, just move it to a separate function.

Reply:
Fixed

Reply:
Fixed